// grunt-glob/Gruntfile.js
module.exports = function(grunt) {
    grunt.initConfig({
        jshint: {
            all: [
                'Gruntfile.js',
                'tasks/*.js',
                '<%= nodeunit.tests %>'
            ],
            options: {
                jshintrc: '.jshintrc'
            }
        },
        clean: {
            test: ['tmp']
        },
        copy: {
            test: {
                files: [{
                    expand: true,
                    cwd: 'test/fixtures',
                    src: [
                        'css/**/*',
                        'js/**/*'
                    ],
                    dest: 'tmp'
                }]
            }
        },
        glob: {
            basic: {
                files: {
                    'tmp/basic.html': 'test/fixtures/basic.html'
                }
            }
        },
        nodeunit: {
            tests: ['test/*_test.js']
        }
    });

    grunt.loadTasks('tasks');

    grunt.loadNpmTasks('grunt-contrib-clean');
    grunt.loadNpmTasks('grunt-contrib-copy');
    grunt.loadNpmTasks('grunt-contrib-jshint');
    grunt.loadNpmTasks('grunt-contrib-nodeunit');

    grunt.registerTask('test', ['clean', 'copy', 'glob', 'nodeunit']);

    grunt.registerTask('default', ['jshint', 'test']);
};
